internal/pkg/pkg.go

```diff
@@ -695,3 +714,47 @@ func SetPkgPathAnnotation(rn *yaml.RNode, pkgPath types.UniquePath) error {
 func RemovePkgPathAnnotation(rn *yaml.RNode) error {
 	return rn.PipeE(yaml.ClearAnnotation(pkgPathAnnotation))
 }
+
+// ReadRGFile returns the resourcegroup object by lazy loading it from the filesytem.
+func (p *Pkg) ReadRGFile(filename string) (*rgfilev1alpha1.ResourceGroup, error) {
+	if p.rgFile == nil {
+		rg, err := ReadRGFile(p.UniquePath.String(), filename)
+		if err != nil {
+			return nil, err
+		}
+		p.rgFile = rg
+	}
+	return p.rgFile, nil
+}
+
+// ReadRGFile reads the KptFile in the given pkg.
+func ReadRGFile(path, filename string) (*rgfilev1alpha1.ResourceGroup, error) {
```

Comment:
I don't see this function is called else where in the PR, suggest merging the logic inside `pkg.ReadRGFile`

Reply:
Also, since both Kptfile and RGFile are kpt owned, placed under `internal/pkg/pkg.go`, and have very simliar file operations, we should make the Pkg to use shared file operating logic (that is not only for avoiding  
duplicate code, but more about make the file functions themselves stable and the features reliable).

Reply:
`ReadRGFile` is required to be exported as we use them directly in our tests. I agree that it would be nice to consolidate the Kptfile and ResourceGroup file operations into a unified function. That would require a non-trivial refactor of our functions, which may be better done in a separate followup PR?

Reply:
Sg. Please add a `TODO(rquitales):` marker

Reply:
Done

Reply:
qq: Which test do you refer to? I"m wondering how unittest would need functions to be public, especially the function is just "read file". Can we use mock function?

Reply:
`ReadRGFile` and `ReadKptfile` is used in the test_Run test cases. I agree, mocking would be a better/cleaner approach. I think when we refactor the Kptfile and RGFile reading functions, we can also clean up the tests as well!